File: TheOffice/VisualBasic/TheOffice/Map.vb
Public Class Map

    Public Sub Display(Rooms As List(Of List(Of Room)))
        Console.Clear()
        Console.WriteLine()
        PrintHorizontalDivider()
        For i = 0 To 4
            PrintAisle(Rooms(i))
        Next
    End Sub

    Private Sub PrintAisle(ByVal Aisle As List(Of Room))

        For i = 0 To 3

            Console.Write(" ")

            For j = 0 To 4

                Dim room As Room = Aisle(j)
                Dim workers As List(Of Worker) = room.Workers

                If i < workers.Count Then
                    Console.Write("| " & workers(i).Name.PadRight(7) & " ")
                Else
                    Console.Write("|         ")
                End If

            Next

            Console.WriteLine("|")

        Next

        PrintHorizontalDivider()

    End Sub

    Private Sub PrintHorizontalDivider()
        Console.WriteLine(" +---------+---------+---------+---------+---------+")
    End Sub

End Class

File: The Office/CSharp/TheOffice/Map.cs
using System;
using System.Collections.Generic;

namespace TheOffice
{
    public class Map
    {
        public void Display(List<List<Room>> rooms)
        {
            Console.Clear();
            Console.WriteLine();
            printHorizontalDivider();
            printRooms(rooms);
        }

        private void printRooms(List<List<Room>> rooms)
        {
            for (int i = 0; i < 5; i++)
            {
                printAisle(rooms[i]);
            }
        }

        private void printAisle(List<Room> aisle)
        {
            for (int i = 0; i < 4; i++)
            {
                Console.Write(" ");

                for (int j = 0; j < 5; j++)
                {
                    Room room = aisle[j];
                    List<Worker> workers = room.Workers;

                    if (i < workers.Count)
                    {
                        Console.Write("| " + workers[i].Name.PadRight(7) + " ");
                    }
                    else
                    {
                        Console.Write("|         ");
                    }
                }

                Console.WriteLine("|");
            }

            printHorizontalDivider();
        }

        private void printHorizontalDivider()
        {
            Console.WriteLine(" +---------+---------+---------+---------+---------+");
        }
    }
}